### PR TITLE
[Azurestack][ACR][AKS] Adding Tests and Fixing Authorization api-version for hybrid profile

### PR DESCRIPTION
**Pull request by Bhuvaneswari-Santharam**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by bganapa — APPROVED** [review]

**Comment by yonzhan** [comment]
Azurestack

**Comment by Juliehzl** [comment]
please make CI pass first

**Comment by Bhuvaneswari-Santharam** [comment]
@Juliehzl  -  All CI checks passed. Please review

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def default_api_version(self):
         ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS: '2016-06-01',
         ResourceType.MGMT_RESOURCE_TEMPLATESPECS: '2015-01-01',
         ResourceType.MGMT_NETWORK_DNS: '2016-04-01',
-        ResourceType.MGMT_AUTHORIZATION: SDKProfile('2016-09-01', {
+        ResourceType.MGMT_AUTHORIZATION: SDKProfile('2015-07-01', {
```
Comment (Juliehzl) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:244]: Is there a downgrade for authorization?

Comment (bganapa) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:244]: This is not a downgrade, This is a bug fix. In our earlier PR we have mistakenly updated the api version @Juliehzl

File: src/azure-cli/azure/cli/command_modules/acr/check_health.py [hunk 2]
```diff
@@ -310,23 +311,25 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
...
-                    if v.client_id == client_id:
-                        from msrestazure.azure_exceptions import CloudError
-                        try:
-                            valid_identity = (resolve_identity_client_id(cmd.cli_ctx, k) == client_id)
-                        except CloudError:
-                            pass
-        if not valid_identity:
-            from ._errors import CMK_MANAGED_IDENTITY_ERROR
-            _handle_error(CMK_MANAGED_IDENTITY_ERROR.format_error_message(registry_name), ignore_errors)
+    if cmd.supported_api_version(min_api='2020-11-01-preview', resource_type=ResourceType.MGMT_CONTAINERREGISTRY):  # pylint: disable=too-many-nested-blocks
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acr/check_health.py:314]: remember to specify operation_group if it is not the default one

File: src/azure-cli/azure/cli/command_modules/acr/check_health.py [hunk 3]
```diff
@@ -310,23 +311,25 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
...
+        # CMK settings
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acr/check_health.py:315]: please also add acr code owner to review your change

Comment (adewaleo) [reply src/azure-cli/azure/cli/command_modules/acr/check_health.py:315]: @Bhuvaneswari-Santharam I tested this locally and it looks good. @northtyphoon we should consider adding some unit tests for `acr check-health`

**Review by adewaleo — APPROVED** [review]
Approved for ACR.
@Bhuvaneswari-Santharam I tested this locally and it looks good. 
@northtyphoon we should consider adding some unit tests for `acr check-health`

**Comment by yonzhan** [comment]
@northtyphoon please add more unit tests as @adewaleo mentioned.

**Comment by adewaleo** [comment]
> @northtyphoon please add more unit tests as @adewaleo mentioned.

@yonzhan, I was making this suggestion as a follow up item. Given `acr check-health` (convenience command for diagnosis) doesn't currently have unit tests. I did manually test to see that this change doesn't break `acr check-health`. I won't want this request to block @Bhuvaneswari-Santharam

**Review by Juliehzl — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_availability_zones_msi.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      string: "{\n  \"code\": \"QuotaExceeded\",\n  \"message\": \"Provisioning of
+        resource(s) for container service cliakstest000001 in resource group acscliapi2
+        failed. Message: Operation could not be completed as it results in exceeding
+        approved Total Regional Cores quota. Additional details - Deployment Model:
+        Resource Manager, Location: westus2, Current Limit: 20, Current Usage: 20,
+        Additional Required: 6, (Minimum) New Limit Required: 26. Submit a request
+        for Quota increase at https://aka.ms/ProdportalCRP/?#create/Microsoft.Support/Parameters/%7B%22subId%22:%228a4f688f-f361-4729-a197-1a8d663b603b%22,%22pesId%22:%2206bfd9d3-516b-d5c6-5802-169c800dec89%22,%22supportTopicId%22:%22e12e3d1d-7fa0-af33-c6d0-3c50df9658a3%22%7D
+        by specifying parameters listed in the \u2018Details\u2019 section for deployment
+        to succeed. Please read more about quota limits at https://docs.microsoft.com/en-us/azure/azure-supportability/regional-quota-requests..
+        Details: \"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_availability_zones_msi.yaml:90]: It seems that this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad.yaml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      string: "{\n  \"code\": \"QuotaExceeded\",\n  \"message\": \"Provisioning of
+        resource(s) for container service cliakstest000001 in resource group acscliapi2
+        failed. Message: Operation could not be completed as it results in exceeding
+        approved Total Regional Cores quota. Additional details - Deployment Model:
+        Resource Manager, Location: eastus, Current Limit: 20, Current Usage: 20,
+        Additional Required: 2, (Minimum) New Limit Required: 22. Submit a request
+        for Quota increase at https://aka.ms/ProdportalCRP/?#create/Microsoft.Support/Parameters/%7B%22subId%22:%228a4f688f-f361-4729-a197-1a8d663b603b%22,%22pesId%22:%2206bfd9d3-516b-d5c6-5802-169c800dec89%22,%22supportTopicId%22:%22e12e3d1d-7fa0-af33-c6d0-3c50df9658a3%22%7D
+        by specifying parameters listed in the \u2018Details\u2019 section for deployment
+        to succeed. Please read more about quota limits at https://docs.microsoft.com/en-us/azure/azure-supportability/regional-quota-requests..
+        Details: \"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad.yaml:93]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad_msi.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      string: "{\n  \"code\": \"QuotaExceeded\",\n  \"message\": \"Provisioning of
+        resource(s) for container service cliakstest000001 in resource group acscliapi2
+        failed. Message: Operation could not be completed as it results in exceeding
+        approved Total Regional Cores quota. Additional details - Deployment Model:
+        Resource Manager, Location: eastus, Current Limit: 20, Current Usage: 20,
+        Additional Required: 2, (Minimum) New Limit Required: 22. Submit a request
+        for Quota increase at https://aka.ms/ProdportalCRP/?#create/Microsoft.Support/Parameters/%7B%22subId%22:%228a4f688f-f361-4729-a197-1a8d663b603b%22,%22pesId%22:%2206bfd9d3-516b-d5c6-5802-169c800dec89%22,%22supportTopicId%22:%22e12e3d1d-7fa0-af33-c6d0-3c50df9658a3%22%7D
+        by specifying parameters listed in the \u2018Details\u2019 section for deployment
+        to succeed. Please read more about quota limits at https://docs.microsoft.com/en-us/azure/azure-supportability/regional-quota-requests..
+        Details: \"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad_msi.yaml:93]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas.yaml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      string: "{\n  \"code\": \"QuotaExceeded\",\n  \"message\": \"Provisioning of
+        resource(s) for container service cliakstest000001 in resource group acscliapi2
+        failed. Message: Operation could not be completed as it results in exceeding
+        approved Total Regional Cores quota. Additional details - Deployment Model:
+        Resource Manager, Location: westus2, Current Limit: 20, Current Usage: 20,
+        Additional Required: 2, (Minimum) New Limit Required: 22. Submit a request
+        for Quota increase at https://aka.ms/ProdportalCRP/?#create/Microsoft.Support/Parameters/%7B%22subId%22:%228a4f688f-f361-4729-a197-1a8d663b603b%22,%22pesId%22:%2206bfd9d3-516b-d5c6-5802-169c800dec89%22,%22supportTopicId%22:%22e12e3d1d-7fa0-af33-c6d0-3c50df9658a3%22%7D
+        by specifying parameters listed in the \u2018Details\u2019 section for deployment
+        to succeed. Please read more about quota limits at https://docs.microsoft.com/en-us/azure/azure-supportability/regional-quota-requests..
+        Details: \"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas.yaml:91]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas_msi.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      strict-transport-security:
+      - max-age=31536000; includeSubDomains
+      vary:
+      - Accept-Encoding
+      x-content-type-options:
+      - nosniff
+    status:
+      code: 200
+      message: OK
+version: 1
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas_msi.yaml:45]: This recording looks incomplete.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_cluster_with_apiserver_authorized_ranges_then_update.yaml [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+      string: "{\n  \"code\": \"QuotaExceeded\",\n  \"message\": \"Provisioning of
+        resource(s) for container service cliakstest000001 in resource group acsclifixapi3
+        failed. Message: Operation could not be completed as it results in exceeding
+        approved Total Regional Cores quota. Additional details - Deployment Model:
+        Resource Manager, Location: westus2, Current Limit: 20, Current Usage: 20,
+        Additional Required: 2, (Minimum) New Limit Required: 22. Submit a request
+        for Quota increase at https://aka.ms/ProdportalCRP/?#create/Microsoft.Support/Parameters/%7B%22subId%22:%228a4f688f-f361-4729-a197-1a8d663b603b%22,%22pesId%22:%2206bfd9d3-516b-d5c6-5802-169c800dec89%22,%22supportTopicId%22:%22e12e3d1d-7fa0-af33-c6d0-3c50df9658a3%22%7D
+        by specifying parameters listed in the \u2018Details\u2019 section for deployment
+        to succeed. Please read more about quota limits at https://docs.microsoft.com/en-us/azure/azure-supportability/regional-quota-requests..
+        Details: \"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_cluster_with_apiserver_authorized_ranges_then_update.yaml:362]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_msi.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      strict-transport-security:
+      - max-age=31536000; includeSubDomains
+      vary:
+      - Accept-Encoding
+      x-content-type-options:
+      - nosniff
+    status:
+      code: 200
+      message: OK
+version: 1
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_msi.yaml:45]: This recording looks incomplete.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_with_skip_role_assignment.yaml [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+        azure-mgmt-containerservice/11.2.0 Azure-SDK-For-Python AZURECLI/2.23.0
+      accept-language:
+      - en-US
+    method: PUT
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/acscliapi3/providers/Microsoft.ContainerService/managedClusters/cliakstest000002?api-version=2020-11-01
+  response:
+    body:
+      string: "{\n  \"code\": \"InvalidParameter\",\n  \"message\": \"The Agentpool
+        nodepool1's vnet clivnet000003 and the cluster are not in the same region.
+        Vnet region is eastus while cluster region is westus2.\"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_with_skip_role_assignment.yaml:310]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_slb_vmss_with_default_mgd_outbound_ip_then_update.yaml [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      accept-language:
+      - en-US
+    method: PUT
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/acscliapi8/providers/Microsoft.ContainerService/managedClusters/cliakstest000001?api-version=2020-11-01
+  response:
+    body:
+      string: "{\n  \"code\": \"ServicePrincipalNotFound\",\n  \"message\": \"Service
+        principal clientID: http://clitest000001 not found in Active Directory tenant
+        657a2bb7-9a58-4be7-87f3-643bb3b7cbb5, Please see https://aka.ms/aks-sp-help
+        for more details.\"\n }"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_slb_vmss_with_default_mgd_outbound_ip_then_update.yaml:85]: It looks like this test case is not executed correctly.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_fqdn_subdomain.yaml [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+      x-content-type-options:
+      - nosniff
+      x-ms-ratelimit-remaining-subscription-resource-requests:
+      - '499'
+      x-powered-by:
+      - ASP.NET
+    status:
+      code: 200
+      message: OK
+version: 1
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_fqdn_subdomain.yaml:154]: This recording looks incomplete.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service.yaml [hunk 14]
```diff
@@ -0,0 +1,450 @@
+interactions:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service.yaml:1]: As a reminder, openshift commands are not well supported.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service_no_aad.yaml [hunk 15]
```diff
@@ -0,0 +1,274 @@
+interactions:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service_no_aad.yaml:1]: As a reminder, openshift commands are not well supported.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_service_no_wait.yaml [hunk 16]
```diff
@@ -0,0 +1,451 @@
+interactions:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_service_no_wait.yaml:1]: As a reminder, openshift commands are not well supported.

**Review by FumingZhang — CHANGES_REQUESTED** [review]
Need to update some recording files, since some of the tests failed.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_service_no_wait.yaml (hunk 16 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_service_no_wait.yaml]: Removed Openshift commands in hybrid profile test

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service.yaml (hunk 14 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_openshift_create_default_service.yaml]: Removed Openshift commands in hybrid profile test

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_availability_zones_msi.yaml (hunk 4 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_availability_zones_msi.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad_msi.yaml (hunk 6 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad_msi.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas_msi.yaml (hunk 8 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas_msi.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_cluster_with_apiserver_authorized_ranges_then_update.yaml (hunk 9 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_cluster_with_apiserver_authorized_ranges_then_update.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_msi.yaml (hunk 10 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_default_service_msi.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas.yaml (hunk 7 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_blb_vmas.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad.yaml (hunk 5 above)
Comment (Bhuvaneswari-Santharam) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/recordings/test_aks_create_aadv1_and_update_with_managed_aad.yaml]: Removed test not supported in Azurestack

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_acs_client.py [hunk 17]
```diff
@@ -0,0 +1,111 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+import unittest
+import mock
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_acs_client.py:8]: `mock` module is deprecated in azure cli. see https://github.com/Azure/azure-cli/pull/19024

**Review by FumingZhang — APPROVED** [review]
AKS test recordings looks good to me.

**Review by Juliehzl — CHANGES_REQUESTED** [review]
`mock` module is deprecated in azure cli. see #19024

File: src/azure-cli/azure/cli/command_modules/acr/tests/hybrid_2020_09_01/test_acr_commands_mock.py [hunk 18]
```diff
@@ -0,0 +1,695 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+try:
+    from urllib.parse import urlencode
+except ImportError:
+    from urllib import urlencode
+import json
+import unittest
+import mock
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acr/tests/hybrid_2020_09_01/test_acr_commands_mock.py:12]: still have mock here.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**